[Model Validation] Finishing touches and documentation (#6577)
mlflow/mlflow: examples/evaluation/README.md

### MLflow evaluation Examples

> The examples in this directory illustrate how you can use the `mlflow.evaluate` API. Specifically,

Reviewer: ```suggestion
The examples in this directory demonstrate how to use the `mlflow.evaluate()` API. Specifically,
```
